### PR TITLE
Basic script pre-loading

### LINKED ISSUE #21: Expose code to level api
When defining levels we are able to define terrain features and starting player/mob positions. I move that we should also have an api for defining starting code. 

While this would introduce some inconsistencies, specifically in the mob spawning api (spawning mobs via the level api vs spawning them using api code run on level load) I think it would benefit both users of craft-engine and level designers overall.

For level designers it would allow them to hardcode mob ai, as well as set up block interactions. 

For end users it would allow easy modding, just load a level and modify its code. It would also let users gain more familiarity with the level’s they are playing without also compromising the original play style of craft.

### PR DESCRIPTION
**Pull request by Alvarito050506**
Expose user/designer's scripts to the level/game controller/API. Fixed a few things with the build system and the demo too.

Closes #21.

### CODE REVIEW

File: src/js/game/API/CodeOrgAPI.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
 function getCodeOrgAPI(controller) {
 	return {
 		/**
-     * Called before a list of user commands will be issued.
-     */
+		 * Called before a list of user commands will be issued.
```
Comment (nobody5050) [inline src/js/game/API/CodeOrgAPI.js:4]: Oh my goodness did you actually change indentation on comments?! You seem to have dropped your 👑 my king

File: src/js/game/GameController.js [hunk 2]
```diff
@@ -53,10 +53,8 @@ class GameController {
 		this.audioPlayer = gameControllerConfig.audioPlayer;
 		this.afterAssetsLoaded = gameControllerConfig.afterAssetsLoaded;
 		this.assetLoader = new AssetLoader(this);
-		this.earlyLoadAssetPacks =
-      gameControllerConfig.earlyLoadAssetPacks || [];
-		this.earlyLoadNiceToHaveAssetPacks =
-      gameControllerConfig.earlyLoadNiceToHaveAssetPacks || [];
+		this.earlyLoadAssetPacks = gameControllerConfig.earlyLoadAssetPacks || [];
+		this.earlyLoadNiceToHaveAssetPacks = gameControllerConfig.earlyLoadNiceToHaveAssetPacks || [];
```
Comment (nobody5050) [inline src/js/game/GameController.js:57]: These are horrible names lmao

**Review by nobody5050 — APPROVED** [review]
Looks good!

File: src/js/game/GameController.js (hunk 2 above)
Comment (Alvarito050506) [reply src/js/game/GameController.js:57]: ~~Not our fault.~~ Technically our fault for not refactoring.

File: src/js/game/API/CodeOrgAPI.js (hunk 1 above)
Comment (Alvarito050506) [reply src/js/game/API/CodeOrgAPI.js:4]: > It looks like a meld between a refactor and api stuff. From the code I can understand this commit looks good

Yeah, sorry about this, I tend to forgot which changes I made and then make big commits.

**Comment by nobody5050** [comment]
I was gonna wait for another review but let’s just merge right now.

**Comment by Alvarito050506** [comment]
Ok, let me squash this so I can make a more descriptive commit name.

**Pull request merged**